### PR TITLE
Extend Seth JSON-RPC API

### PR DESCRIPTION
**Pull request by knkski**
Proposes extensions to the Seth JSON-RPC API that allow it to handle functionality that currently requires talking to the Sawtooth REST API.

Signed-off-by: Kenneth Koski <knkski@bitwise.io>

### CODE REVIEW

**Comment by knkski** [comment]
@TomBarnes: I clarified in the Prior art section which two JSON-RPC API extensions we conform to:

https://github.com/hyperledger/sawtooth-rfcs/pull/22/commits/eeeb62e8065763ed7a61cc63456c76d8620281d2

**Comment by knkski** [comment]
Propose this RFC be merged.

After approval by the core team, this RFC will enter its final comment period.
* [ ] Adam Ludvik
* [ ] Andi Gunderson
* [ ] Dan Middleton
* [ ] James Mitchell
* [ ] Peter Schwarz
* [ ] Shawn Amundson

File: text/0000-seth-rpc-extensions.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+## Contract Chaining
+
+Sawtooth has the concept of a limited set of input and output address that a
+transaction is limited to reading from/writing to. That concept doesn't exist
+in Ethereum, and makes the concept of contract chaining (e.g. creating a
+contract from another contract) difficult. The `eth_sendTransaction` endpoint
+will be extended to allow an argument that sets the list of inputs and outputs
+to be the entire Sawtooth Seth address space, enabling general contract
+chaining.
```
Comment (dcmiddle) [inline text/0000-seth-rpc-extensions.md]: Can you expand on this? How do inputs/outputs default today? What implications are there in enabling the entire address space ... performance? ... security? How do the transaction input/output addresses interact or not with permissions?

Comment (knkski) [reply text/0000-seth-rpc-extensions.md]: After talking with @aludvik, I think we can default to allowing contract chaining (i.e., transactions are simply limited to the seth address namespace). This is what Ethereum-based tools expect, and we can then tackle restricting the address space for contract chaining as a separate ticket. So, this section has been removed.

**Comment by dcmiddle** [comment]
@knkski can you please notify the mail list of the FCP

**Comment by knkski** [comment]
Due to changes to the RFC, calling again for a move to FCP. The change was removing a section about contract chaining. After approval by the core team, this RFC will enter its final comment period.

* [ ] Adam Ludvik
* [ ] Andi Gunderson
* [x] Dan Middleton
* [ ] James Mitchell
* [ ] Peter Schwarz
* [ ] Shawn Amundson

**Comment by vaporos** [comment]
This RFC has entered the final comment period. If there are no substantive requests for changes by November 9, 2018 it will be merged.

**Pull request merged**